packages/taiko-client/bindings/encoding/input.go

```diff
@@ -423,6 +430,47 @@ func EncodeProveBlockInput(
 	return b, nil
 }
 
+// EncodeProveBlocksInput performs the solidity `abi.encode` for the given TaikoL1.proveBlocks input.
+func EncodeProveBlocksInput(
+	metas []metadata.TaikoBlockMetaData,
+	transitions []bindings.TaikoDataTransition,
+) ([][]byte, error) {
+	var (
+		b   [][]byte
+		err error
+	)
+	if len(metas) != len(transitions) {
```

Comment:
Move this code before L438 can be better.

Reply:
ok